core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -721,6 +722,28 @@ func (vdb *VersionedDB) Close() {
 	// no need to close db since a shared couch instance is used
 }
 
+// Remove removes the couch dbs and redologger db for the channel.
+// It is not an error if a database does not exist.
+func (vdb *VersionedDB) Remove() error {
```

Comment:
Won't this be more intuitive if you have this function on the provider? Here it may be reusing some code but it's over using as well. This requires the construction of db object and the current implementation of that function creates the db, checks for redo logs etc. For instance, for a non-existent db, it will actually create one and then delete it.